src/bio/bio_monitor.c

```diff
@@ -178,22 +250,30 @@ bio_dev_set_faulty(struct bio_xs_context *xs)
 }
 
 static inline struct bio_dev_health *
-xs_ctxt2dev_health(struct bio_xs_context *ctxt)
+xs_ctxt2dev_health(struct bio_xs_context *ctxt, enum smd_dev_type st)
 {
 	D_ASSERT(ctxt != NULL);
+	struct bio_xs_blobstore *bxb;
+
+	bxb = bio_xs_context2xs_blobstore(ctxt, st);
 	/* bio_xsctxt_free() is underway */
-	if (ctxt->bxc_blobstore == NULL)
+	if (bxb == NULL)
 		return NULL;
 
-	return &ctxt->bxc_blobstore->bb_dev_health;
+	return &bxb->bxb_blobstore->bb_dev_health;
 }
 
+struct get_spdk_cb_arg {
+	struct bio_xs_context	*ctxt;
+	enum smd_dev_type	 st;
+};
```

Comment:
I suggest we don't support health monitor for WAL & Meta device for this moment, let's assume health monitor is on data device only.

Reply:
ok!